csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -302,42 +335,47 @@ private IArrowType GetType(TableFieldSchema field, IArrowType type)
             return type;
         }
 
-        private IArrowReader? ReadChunkWithRetries(TokenProtectedReadClientManger clientMgr, string streamName)
+        private IArrowReader? ReadChunkWithRetries(TokenProtectedReadClientManger clientMgr, string streamName, Activity? activity)
         {
-            Func<Task<IArrowReader?>> func = () => Task.FromResult<IArrowReader?>(ReadChunk(clientMgr, streamName));
-            return RetryManager.ExecuteWithRetriesAsync<IArrowReader?>(clientMgr, func, MaxRetryAttempts, RetryDelayMs).GetAwaiter().GetResult();
+            Func<Task<IArrowReader?>> func = () => Task.FromResult<IArrowReader?>(ReadChunk(clientMgr, streamName, activity));
+            return RetryManager.ExecuteWithRetriesAsync<IArrowReader?>(clientMgr, func, activity, MaxRetryAttempts, RetryDelayMs).GetAwaiter().GetResult();
         }
 
-        private static IArrowReader? ReadChunk(TokenProtectedReadClientManger clientMgr, string streamName)
+        private static IArrowReader? ReadChunk(TokenProtectedReadClientManger clientMgr, string streamName, Activity? activity)
         {
-            return ReadChunk(clientMgr.ReadClient, streamName);
+            return ReadChunk(clientMgr.ReadClient, streamName, activity);
         }
 
-        private static IArrowReader? ReadChunk(BigQueryReadClient client, string streamName)
+        private static IArrowReader? ReadChunk(BigQueryReadClient client, string streamName, Activity? activity)
         {
             // Ideally we wouldn't need to indirect through a stream, but the necessary APIs in Arrow
             // are internal. (TODO: consider changing Arrow).
+            activity?.AddTag("ReadStream", streamName);
```

Comment:
You might consider tag names from this namespace ...
https://opentelemetry.io/docs/specs/semconv/registry/attributes/messaging/